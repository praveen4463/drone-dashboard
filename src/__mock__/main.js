"use strict";

let data = {
	devices: [
		{
			deviceUID: "drone1",
			found: true,
			coordinates: "1,1",
			error: null
		},
		{
			deviceUID: "drone2",
			found: true,
			coordinates: "2,2",
			error: null
		},
		{
			deviceUID: "drone1",
			found: false,
			coordinates: null,
			error: null
		},
		{
			deviceUID: "drone3",
			found: true,
			coordinates: "3,3"
			error: null
		}
	],
	json: () => Promise.resolve(this)
}

export default const fetch = jest.fn().mockImplementation((url, options) => Promise.resolve(data));