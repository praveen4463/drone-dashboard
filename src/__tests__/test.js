'use strict';

import React from "react";
import {Footer, DevicesDetail, Dashboard} from "../main";
import renderer from "react-test-renderer";


describe("Dashboard tests", () => {
	it("Footer renders correctly", () => {
		const tree = renderer.create(<Footer refreshTime="11"></Footer>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("DevicesDetail renders correctly with row formatting", () => {
		let dummyDevicesDetail = [], obj = {};

		obj.deviceUID = "device1";
		obj.found = true;
		obj.currentCoordinates = "1,1";
		obj.previousCoordinates = ".1,.1";
		obj.moved = 0;
		obj.speed = 0;
		dummyDevicesDetail.push(obj);

		obj = {};
		obj.deviceUID = "device2";
		obj.found = true;
		obj.currentCoordinates = "2,2";
		obj.previousCoordinates = ".2,.2";
		obj.moved = 4;
		obj.speed = 40;
		dummyDevicesDetail.push(obj);

		obj = {};
		obj.deviceUID = "device3";
		obj.found = false;
		obj.currentCoordinates = null;
		obj.previousCoordinates = null;
		obj.moved = 0;
		obj.speed = 0;
		dummyDevicesDetail.push(obj);

		const tree = renderer.create(<DevicesDetail devices={dummyDevicesDetail}></DevicesDetail>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	test("Dashboard renders correctly", done => {
		const data = {
			devices: [
				{
					deviceUID: "drone1",
					found: true,
					coordinates: "1,1",
					error: null
				},
				{
					deviceUID: "drone2",
					found: true,
					coordinates: "2,2",
					error: null
				},
				{
					deviceUID: "drone1",
					found: false,
					coordinates: null,
					error: null
				},
				{
					deviceUID: "drone3",
					found: true,
					coordinates: "3,3",
					error: null
				}
			],
			json: () => Promise.resolve(data)
		};
		window.fetch = jest.fn().mockImplementation((url, options) => Promise.resolve(data));
		const tree = renderer.create(<Dashboard></Dashboard>);
		setTimeout(() => { 
			tree.update(<Dashboard></Dashboard>);
			expect(tree.toJSON()).toMatchSnapshot();
			done();
	 	}, 1000);
	});
});