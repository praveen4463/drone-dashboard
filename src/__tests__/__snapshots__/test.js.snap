// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard tests Dashboard renders correctly 1`] = `
<div>
  <div>
    <span>
      <h1>
        Device Dashboard
      </h1>
    </span>
  </div>
  <div
    className="error"
  >
    <span />
  </div>
  <div
    className="project"
  >
    <span>
      Displaying all available devices for ProjectID = 
      999
    </span>
  </div>
  <div
    className="main"
  >
    <table>
      <tbody>
        <tr>
          <td>
            Device Type
          </td>
          <td>
            Device UID
          </td>
          <td>
            Online Status
          </td>
          <td>
            Current coordinates
          </td>
          <td>
            Previous coordinates
          </td>
          <td>
            Moved After Last contact (kms)
          </td>
          <td>
            Current Speed (kms/hr)
          </td>
        </tr>
        <tr
          className=""
        >
          <td>
            <img
              alt="drone"
              height="36px"
              src="drone.jpg"
              width="36px"
            />
          </td>
          <td>
            drone1
          </td>
          <td>
            Online
          </td>
          <td>
            1,1
          </td>
          <td />
          <td>
            0
          </td>
          <td>
            0
          </td>
        </tr>
        <tr
          className=""
        >
          <td>
            <img
              alt="drone"
              height="36px"
              src="drone.jpg"
              width="36px"
            />
          </td>
          <td>
            drone2
          </td>
          <td>
            Online
          </td>
          <td>
            2,2
          </td>
          <td />
          <td>
            0
          </td>
          <td>
            0
          </td>
        </tr>
        <tr
          className="offline"
        >
          <td>
            <img
              alt="drone"
              height="36px"
              src="drone.jpg"
              width="36px"
            />
          </td>
          <td>
            drone1
          </td>
          <td>
            Offline
          </td>
          <td />
          <td />
          <td>
            0
          </td>
          <td>
            0
          </td>
        </tr>
        <tr
          className=""
        >
          <td>
            <img
              alt="drone"
              height="36px"
              src="drone.jpg"
              width="36px"
            />
          </td>
          <td>
            drone3
          </td>
          <td>
            Online
          </td>
          <td>
            3,3
          </td>
          <td />
          <td>
            0
          </td>
          <td>
            0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="footer"
  >
    <span>
      Devices will be contated  and updated automatically after 
      11
       seconds.
    </span>
    <br />
    <span>
      Offline Devices will have a line through on their row and in red.
    </span>
    <br />
    <span>
      Devices not moving for more than 10 secs will be highlighted in red background.
    </span>
    <br />
  </div>
</div>
`;

exports[`Dashboard tests DevicesDetail renders correctly with row formatting 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        Device Type
      </td>
      <td>
        Device UID
      </td>
      <td>
        Online Status
      </td>
      <td>
        Current coordinates
      </td>
      <td>
        Previous coordinates
      </td>
      <td>
        Moved After Last contact (kms)
      </td>
      <td>
        Current Speed (kms/hr)
      </td>
    </tr>
    <tr
      className="highlight"
    >
      <td>
        <img
          alt="drone"
          height="36px"
          src="drone.jpg"
          width="36px"
        />
      </td>
      <td>
        device1
      </td>
      <td>
        Online
      </td>
      <td>
        1,1
      </td>
      <td>
        .1,.1
      </td>
      <td>
        0
      </td>
      <td>
        0
      </td>
    </tr>
    <tr
      className=""
    >
      <td>
        <img
          alt="drone"
          height="36px"
          src="drone.jpg"
          width="36px"
        />
      </td>
      <td>
        device2
      </td>
      <td>
        Online
      </td>
      <td>
        2,2
      </td>
      <td>
        .2,.2
      </td>
      <td>
        4
      </td>
      <td>
        40
      </td>
    </tr>
    <tr
      className="offline"
    >
      <td>
        <img
          alt="drone"
          height="36px"
          src="drone.jpg"
          width="36px"
        />
      </td>
      <td>
        device3
      </td>
      <td>
        Offline
      </td>
      <td />
      <td />
      <td>
        0
      </td>
      <td>
        0
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Dashboard tests Footer renders correctly 1`] = `
<div
  className="footer"
>
  <span>
    Devices will be contated  and updated automatically after 
    11
     seconds.
  </span>
  <br />
  <span>
    Offline Devices will have a line through on their row and in red.
  </span>
  <br />
  <span>
    Devices not moving for more than 10 secs will be highlighted in red background.
  </span>
  <br />
</div>
`;
